fault-proof/tests/common/contracts.rs

```diff
@@ -0,0 +1,263 @@
+    superchain_config::SuperchainConfig,
+};
+use tracing::{debug, info};
+
+use fault_proof::{L1Provider, L2Provider, L2ProviderTrait};
+
+use super::constants::{
+    AGGREGATION_VKEY, AIRGAP_PERIOD, CHALLENGER_ADDRESS, CHALLENGER_BOND, DEPLOYER_ADDRESS,
+    FALLBACK_TIMEOUT, INIT_BOND, MAX_CHALLENGE_DURATION, MAX_PROVE_DURATION, PROPOSER_ADDRESS,
+    RANGE_VKEY_COMMITMENT, ROLLUP_CONFIG_HASH, TEST_GAME_TYPE,
+};
+
+/// Container for deployed contracts
+pub struct DeployedContracts {
+    pub factory: Address,
+    pub portal: Address,
+    pub access_manager: Address,
+    pub game_implementation: Address,
+}
+
+/// Deploy all contracts required for E2E testing
+pub async fn deploy_test_contracts(
+    provider: L1Provider,
+    l2_provider: L2Provider,
+) -> Result<DeployedContracts> {
+    info!("Deploying all contracts for E2E testing");
+
+    // 1. Deploy DisputeGameFactory as proxy (matching production pattern)
+    debug!("Deploying DisputeGameFactory implementation...");
+    let factory_impl = DisputeGameFactory::deploy(provider.clone()).await?;
+    let factory_impl_addr = *factory_impl.address();
+    info!("✓ DisputeGameFactory implementation deployed at: {}", factory_impl_addr);
+
+    // Deploy factory proxy with initialization
+    debug!("Deploying DisputeGameFactory proxy...");
+    let init_data = DisputeGameFactory::initializeCall { _owner: DEPLOYER_ADDRESS }.abi_encode();
+
+    let factory_proxy =
+        ERC1967Proxy::deploy(provider.clone(), factory_impl_addr, init_data.into()).await?;
+    let factory = *factory_proxy.address();
+    info!("✓ DisputeGameFactory proxy deployed at: {}", factory);
+
+    // 2. Deploy MockOptimismPortal2 (needed for AnchorStateRegistry)
+    debug!("Deploying MockOptimismPortal2...");
+    let portal_instance =
+        MockOptimismPortal2::deploy(provider.clone(), TEST_GAME_TYPE, U256::from(AIRGAP_PERIOD))
+            .await?;
+    let portal = *portal_instance.address();
+    info!("✓ MockOptimismPortal2 deployed at: {}", portal);
+
+    // 3. Deploy AnchorStateRegistry as proxy (before AccessManager)
+    debug!("Deploying SuperchainConfig...");
+    let superchain_config = SuperchainConfig::deploy(provider.clone()).await?;
+    let superchain_config_addr = *superchain_config.address();
+    info!("✓ SuperchainConfig deployed at: {}", superchain_config_addr);
+
+    // Deploy AnchorStateRegistry implementation
+    debug!("Deploying AnchorStateRegistry implementation...");
+    let anchor_impl = AnchorStateRegistry::deploy(provider.clone()).await?;
+    let anchor_impl_addr = *anchor_impl.address();
+    info!("✓ AnchorStateRegistry implementation deployed at: {}", anchor_impl_addr);
+
+    // Prepare initialization data with expected test values
+    // NOTE: We use the finalized L2 block number - 100 as the starting anchor root and test 3 games
+    // with proposal interval of 10 blocks.
+    let l2_block_number = U256::from(
```

Comment:
same comment as before, i think op-succinct-infra can be used to detect issues with a live chain, and the e2e test can be used to test the core logic of the proposer / challenger